node/subsystem/src/jaeger.rs

```diff
@@ -0,0 +1,256 @@
+#[inline(always)]
+pub fn pov_span(pov: &PoV, span_name: impl Into<String>) -> JaegerSpan {
+	hash_span(&pov.hash(), span_name)
+}
+
+/// Creates a `Span` referring to the given hash. All spans created with [`hash_span`] with the
+/// same hash (even from multiple different nodes) will be visible in the same view on Jaeger.
+#[inline(always)]
+pub fn hash_span(hash: &Hash, span_name: impl Into<String>) -> JaegerSpan {
+	INSTANCE.read_recursive().span(hash, span_name).into()
+}
+
+/// Stateful convenience wrapper around [`mick_jaeger`].
+pub enum Jaeger {
+	/// Launched and operational state.
+	Launched {
+		/// [`mick_jaeger`] provided API to record spans to.
+		traces_in: Arc<mick_jaeger::TracesIn>,
+	},
+	/// Preparation state with the necessary config to launch the collector.
+	Prep(JaegerConfig),
+	/// Uninitialized, suggests wrong API usage if encountered.
+	None,
+}
+
+impl Jaeger {
+	/// Spawn the jaeger instance.
+	pub fn new(cfg: JaegerConfig) -> Self {
+		Jaeger::Prep(cfg)
+	}
+
+	/// Spawn the background task in order to send the tracing information out via udp
+	#[cfg(target_os = "unknown")]
+	pub fn launch<S: SpawnNamed>(self, _spawner: S) -> result::Result<(), JaegerError> {
+		Ok(())
+	}
+
+	/// Spawn the background task in order to send the tracing information out via udp
+	#[cfg(not(target_os = "unknown"))]
+	pub fn launch<S: SpawnNamed>(self, spawner: S) -> result::Result<(), JaegerError> {
+		let cfg = match self {
+			Self::Prep(cfg) => Ok(cfg),
+			Self::Launched{ .. } => {
+				return Err(JaegerError::AlreadyLaunched)
+			}
+			Self::None => Err(JaegerError::MissingConfiguration),
+		}?;
+
+		let jaeger_agent = cfg.agent_addr;
+
+		log::info!("🐹 Collecting jaeger spans for {:?}", &jaeger_agent);
+
+		let (traces_in, mut traces_out) = mick_jaeger::init(mick_jaeger::Config {
+			service_name: format!("{}-{}", cfg.node_name, cfg.node_name),
+		});
+
+
+		// Spawn a background task that pulls span information and sends them on the network.
+		spawner.spawn("jaeger-collector", Box::pin(async move {
+			let res = async_std::net::UdpSocket::bind("127.0.0.1:0").await
+				.map_err(JaegerError::PortAllocationError);
+			match res {
+				Ok(udp_socket) => loop {
+					let buf = traces_out.next().await;
+					// UDP sending errors happen only either if the API is misused or in case of missing privilege.
+					if let Err(e) = udp_socket.send_to(&buf, jaeger_agent).await
+						.map_err(|e| JaegerError::SendError(e))
+					{
+						log::trace!("Jaeger: {:?}", e);
+					}
+				}
+				Err(e) => {
+					log::warn!("Jaeger: {:?}", e);
+				}
+			}
+		}));
+
+		*INSTANCE.write() = Self::Launched {
+			traces_in,
+		};
+		Ok(())
+	}
+
+	#[inline(always)]
+	fn traces_in(&self) -> Option<&Arc<mick_jaeger::TracesIn>> {
+		match self {
+			Self::Launched {
+				traces_in,
+				..
+			} => Some(&traces_in),
+			_ => None,
+		}
+	}
+
+	fn span(&self, hash: &Hash, span_name: impl Into<String>) -> Option<mick_jaeger::Span> {
```

Comment:
`hash` needs to be a closure. 

That we only call when jaeger is in a `Launched` state. Otherwise we will waste a lot of resource power for calculating hashes (especially from the PoV) when jaeger is disabled.

Reply:
> Otherwise it is okay for now.

Can you elaborate on this part?